pkg/controller/awsprivatelink/cleanup.go

```diff
@@ -171,6 +176,25 @@ func (r *ReconcileAWSPrivateLink) cleanupHostedZone(awsClient awsclient.Client,
 	}
 
 	hzLog := logger.WithField("hostedZoneID", hzID)
+
+	// Determine if hzID is one of SharedHostedZoneDomains
+	var isSharedHostedZone bool
+	zoneResp, err := awsClient.GetHostedZone(&route53.GetHostedZoneInput{
```

Comment:
It's kind of a pity that `findHostedZone` on L168 already gets the information we need (the Name), which would save us this call if we happened to go through that path. IDK if it's worth refactoring `findHostedZone` to return the name as well as the ID, then we can conditionally do this whole chunk only if we need it.

Reply:
ack